### DISCUSSION TITLE
Decentralized storage for media uploads?

### DISCUSSION THREAD

**Discussion by namelessperson0** [deso-protocol/dips#179] [Q&A]
In which servers are the uploaded images and videos stored? 
I am assuming only the resource identifiers of the images and videos are stored in blockchain.
Will using decentralized storage mechanisms like IPFS be useful?

**Comment by tijno** [comment]
images/gifs uploaded with post are stored on google cloud storage
it is also possible to attach/embed arweave & ipfs hosted images
videos are hosting on cloudflare

**Reply by carry2web** [reply]
Check discussion https://github.com/deso-protocol/dips/discussions/249
